AutomatticTracks/src/main/java/com/automattic/android/tracks/TracksClient.java

```diff
@@ -267,6 +274,8 @@ public void run() {
                                         mInsertEventsQueue.notifyAll();
                                     }
                                 }
+                            } else {
+                                lastNetworkErrorTimestamp = 0L;
```

Comment:
Could it be the case that this value is overwritten before it's being compared?

For example:
1. We set the error timestamp to currentTimeMs (262)
2. The bufferCopyThread begins inserting the mustKeepEvents to the DB (88)
3. The network thread processes another event with success, sets lastNetworkErrorTimestamp to 0 (278)
4. sendToWireThread begins processing events in the database (121)
5. lastNetworkErrorTimestamp is 0 -> shouldWait is false (125)
6. There is no pending flush so the events are not enqueued in the network thread (127)

Would be bizarre to have the network thread process another message before the sendToWire thread starts running but it looks like it's possible.

I think this will prevent that with another side-effect that the retry will happen two minutes from the first error instead of the last error.
```
if ((mPendingFlush || (!shouldWait && EventTable.getEventsCount(mContext) > DEFAULT_EVENTS_QUEUE_THREESHOLD))
                                    && NetworkUtils.isNetworkAvailable(mContext)) {
    lastNetworkErrorTimestamp = 0L;
    ...
}

if (lastNetworkErrorTimestamp == 0L && isErrorResponse) {
    lastNetworkErrorTimestamp = System.currentTimeMillis();
    ...
}
```